lib/utils/canonicalize.js

```diff
@@ -0,0 +1,12 @@
+const canonicalize = str => {
+  if (!str) {
+    return str;
+  }
+  return str
+    .toLowerCase()
+    .normalize()
+    .trim()
+    .replace(/\s+/g, "");
```

Comment:
I think to be safe what we want to do here is replace any sequence of non alpha-numeric characters with a single '-', so (pseudo code) something like

```js
.replace(/[^a-z0-9]+/g, '-')
```

Reply:
Hmm, I don't think its uncommon to use symbols in these types of names though. If cohorts are similar to guilds/clans in games, I see symbols used quite often in those, even some that are entirely made up of symbols. Is there a reason users should not be allowed to use symbols? And if so, should we let them know that symbols aren't allowed?

Reply:
> Hmm, I don't think its uncommon to use symbols in these types of names though. If cohorts are similar to guilds/clans in games, I see symbols used quite often in those, even some that are entirely made up of symbols. Is there a reason users should not be allowed to use symbols? And if so, should we let them know that symbols aren't allowed?
> 
> Edit: I see that we aren't sanitizing team names either. Should we? And should those allow symbols or not?

Is good question. I think for this case we're mainly saying "you can't have a cohort name that's too similar to another existing cohort name"

Like we should not allow the name "The Chiefs!?!" if the name "The Chiefs!" already exists. Or what do you think?